src/authentication.ts

```diff
@@ -187,18 +211,23 @@ export default class GitpodAuthenticationProvider extends Disposable implements
 		try {
 			// For the Gitpod scope list, order doesn't matter so we immediately sort the scopes
 			const sortedScopes = scopes.sort();
+			const validScopes = await this.fetchValidScopes();
+			const sortedFilteredScopes = sortedScopes.filter(s => !validScopes || validScopes.includes(s));
+			if (sortedScopes.length !== sortedFilteredScopes.length) {
+				this._logger.warn(`Creating session with only valid scopes ${sortedFilteredScopes.join(',')}, original scopes where ${sortedScopes.join(',')}`);
+			}
```

Comment:
I can filter scopes here before asking to oauth, this way there is only two places we check for valid scopes, so we can avoid having this same logic server side which would make it three places with the same logic, that would be redundant IMHO